### PR TITLE
 CSHARP-4911: Implement single entry point for all driver's extensions.

### PR DESCRIPTION
**Pull request by sanych-sun**
Related to CSHARP-4911: Refactor AWS auth library to a separate package

### CODE REVIEW

File: src/MongoDB.Driver.Authentication.AWS/RegistrationHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+namespace MongoDB.Driver.Authentication.AWS
+{
+    /// <summary>
+    /// Auth Registry Extensions.
+    /// </summary>
+    public static class RegistrationHelpers
```
[the remark this answers is not in the document]
Comment (sanych-sun) [reply src/MongoDB.Driver.Authentication.AWS/RegistrationHelpers.cs]: Done

File: src/MongoDB.Driver.Authentication.AWS/RegistrationHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            extensionManager.SaslMechanisms.Register(AWSSaslMechanism.MechanismName, AWSSaslMechanism.Create);
+            return extensionManager;
+        }
+
+        /// <summary>
+        /// Registers AWS Kms Provider.
+        /// </summary>
+        /// <param name="extensionManager">Extensions Manager.</param>
+        public static IExtensionManager RegisterAWSKmsProvider(this IExtensionManager extensionManager)
```
[the remark this answers is not in the document]
Comment (sanych-sun) [reply src/MongoDB.Driver.Authentication.AWS/RegistrationHelpers.cs]: Done + expose single method only as discussed

**Review by BorisDog — APPROVED** [review]
LGTM.
Please evaluate the idea of additional registries nesting within `IExtensionMananger`.

**Review by adelinowona — APPROVED** [review]
LGTM

**Pull request merged**